### PR TITLE
Update SDK external types

### PR DESCRIPTION
**Pull request by rcoh**
## Motivation and Context
- https://github.com/awslabs/smithy-rs/issues/3019

## Description
Prune / split out allow external types

## Testing
- CI

----

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

File: aws/sdk/sdk-external-types.toml [hunk 1]
```diff
@@ -1,21 +1,30 @@
...
+    "aws_smithy_async::rt::sleep::*",
+    "aws_smithy_async::time::*",
+    "aws_smithy_async::future::pagination_stream::PaginationStream",
+    # only allowed in from impl for presigned request
     "http::request::Request",
-    "http::error::Error",
-    "http::uri::Uri",
-    "http::method::Method",
+
+    ### All below this line need to be fixed:
```
Comment (jdisanti) [inline aws/sdk/sdk-external-types.toml:14]: The event stream one at the very bottom doesn't need to be fixed. It just needs a tooling upgrade.

File: aws/sdk/sdk-external-types.toml [hunk 2]
```diff
@@ -1,21 +1,30 @@
...
-    "aws_smithy_runtime::*",
-    "aws_smithy_runtime_api::*",
     "aws_smithy_types::*",
+    "aws_credential_types::*",
     "aws_types::*",
-
-    "http::header::map::HeaderMap",
-    "http::header::value::HeaderValue",
+    "aws_smithy_runtime_api::*",
+    "aws_smithy_runtime::*",
```
Comment (jdisanti) [inline aws/sdk/sdk-external-types.toml:7]: Since this will be an unstable crate, we should be careful about what gets allowed here.

Comment (rcoh) [reply aws/sdk/sdk-external-types.toml:7]: oh good catch, bad copy-paste by me. It's actually unused.

File: aws/sdk/sdk-external-types.toml (hunk 1 above)
Comment (rcoh) [reply aws/sdk/sdk-external-types.toml:14]: the crate won't go 1.0 though—we might want a newtype?

**Review by Velfi — APPROVED** [review]

**Pull request merged**